src/api/sync.rs

```diff
@@ -163,21 +166,45 @@ impl ApiBuilder {
     pub fn build(self) -> Result<Api, ApiError> {
         let headers = self.build_headers();
 
+        // Create the TLS connector when native-tls feature is enabled
+        #[cfg(feature = "native-tls")]
+        let agent = {
```

Comment:
I think we should factor out this so that there's only 1 compile time flag (just so that it's easier to reason about things).

I can take care of it if you want.

Reply:
> I can take care of it if you want.

Sure, that'd be great Rust isn't my main stack

Reply:
I've done that here: https://github.com/huggingface/hf-hub/commit/b79b16fadf3540b3f920080c14fcf6d5980ffdd1

Happy to put you as a co-author in some related PR to give back credit. Thanks for your contribution.

Reply:
Sure, thank you.